pkg/controller/controller.go

```diff
@@ -1670,11 +1670,15 @@ func (c *controller) getServiceClassPlanAndBrokerForBinding(instance *v1alpha1.I
 // returns an error. If the AuthSecret field is nil, empty values are
 // returned.
 func getAuthCredentialsFromBroker(client kubernetes.Interface, broker *v1alpha1.Broker) (username, password string, err error) {
-	if broker.Spec.AuthSecret == nil {
+	// TODO: when we start supporting additional auth schemes, this code will have to accommodate
+	// the new schemes
+	basicAuthSecret := broker.Spec.AuthInfo.BasicAuthSecret
```

Comment:
this is nil-safe because `AuthInfo` is a `BrokerAuthInfo`, not a `*BrokerAuthInfo`, and there's a check for a `nil` basic auth secret (which is a `*v1.ObjectReference`)